include/Utils/DIPUtils.h

```diff
@@ -84,8 +98,15 @@ void calcAndStoreFMAwTailProcessing(OpBuilder &builder, Location loc,
         Value outputVec = builder.create<MaskedLoadOp>(
             loc, vecType, output, ValueRange{beginIdx, endIdx}, extraElemMask,
             zeroPadding);
-        Value resVec =
-            builder.create<FMAOp>(loc, inputVec, kernelVec, outputVec);
+        Value resVec = {};
+        auto elemTy = vecType.getElementType();
+        if (elemTy.isF32()) {
+          resVec = builder.create<vector::FMAOp>(loc, inputVec, kernelVec, outputVec);
+        } else if (elemTy.isInteger(32)) {
+          Value mulVec = builder.create<arith::MulIOp>(loc, inputVec, kernelVec);
+          resVec = builder.create<arith::AddIOp>(loc, mulVec, outputVec);
+        }
```

Comment:
All of this (101 - 108) can be wrapped up nicely in a utility function for convenience

Reply:
Added